libraries/chain/stake.cpp

```diff
@@ -91,11 +91,16 @@ void update_proxied_traversal(
 
         while ((grant_itr != grants_idx.end()) && (grant_itr->token_code == token_code) && (grant_itr->grantor_name == agent->account)) {
             auto proxy_agent = get_agent(token_code, agents_idx, grant_itr->recipient_name);
-            update_proxied_traversal(ram, now, token_code, agents_idx, grants_idx, proxy_agent, last_reward, votes_changes, force);
-
-            if (proxy_agent->proxy_level < agent->proxy_level &&
-                grant_itr->break_fee >= proxy_agent->fee &&
-                grant_itr->break_min_own_staked <= proxy_agent->min_own_staked)
+            update_proxied_traversal(ram, now, token_code, agents_idx, grants_idx, autorcs_idx, proxy_agent, last_reward, votes_changes, force, custom_autorecall_mode);
+            
+            auto autorc_itr = autorcs_idx.find(agent_key(token_code, agent->account));
+            bool grantor_breaks_due_to_fee = grant_itr->break_fee < proxy_agent->fee && 
+                (!custom_autorecall_mode || (autorc_itr != autorcs_idx.end() && autorc_itr->break_fee_enabled));
```

Comment:
if we use only inverted `!custom_autorecall_mode` then maybe initially invert it and name `legacy_mode` or so?
each removed `!` makes long bool expressions slightly simpler to read ; )